icu4j/main/classes/core/src/com/ibm/icu/impl/breakiter/CjkBreakEngine.java

```diff
@@ -211,18 +268,45 @@ public int divideUpDictionaryRange(CharacterIterator inText, int startPos, int e
         int correctedNumBreaks = 0;
         for (int i = numBreaks - 1; i >= 0; i--) {
             int pos = charPositions[t_boundary[i]] + startPos;
-            if (!(foundBreaks.contains(pos) || pos == startPos)) {
+            // In phrase breaking, there has to be a breakpoint between Cj character and close
+            // punctuation.
+            // E.g.［携帯電話］正しい選択 -> ［携帯▁電話］▁正しい▁選択 -> breakpoint between ］ and 正
+            if (!foundBreaks.contains(pos)
+                    && (pos != startPos
+                    || (isPhraseBreaking && pos == startPos && pos > 0
```

Comment:
I believe you do NOT need to have "&& pos == startPos"  because that is ALWAYS TRUE after "pos != startPos  || " You only need 
if (!foundBreaks.contains(pos)
                    && (pos != startPos
                    || (isPhraseBreaking && pos > 0
                           && fClosePunctuationSet.contains(inText.setIndex(pos - 1))))) {

Reply:
Resolved